sjb/syntax.md

```diff
@@ -0,0 +1,211 @@
+or one that inherits from another. A standalone job can declare options; a job
+with a parent can extend or override any option from the parent.
+
+Parent-child relationships can nest -- to generate a job, the topmost parent
+configuration is loaded, then iteratively extended and overridden by children
+configurations until the bottommost configuration has been evaluated.
+
+## `parameters`
+
+`parameters` is an optional field which adds parameters to the Jenkins job. The
+field contains a list of parameter definitions. The parameter definition syntax
+is:
+
+```yaml
+parameters:
+  - name: ""          # environment variable name
+    description: ""   # human-readable description
+    default_value: "" # default value [optional]
+```
+
+## `provision`
+
+`provision` is a required field which gives the options for provisioning a VM
+in AWS EC2 for the job. The syntax is:
+
+```yaml
+provision:
+  os: ""       # operating system. only 'rhel' is supported
+  stage: ""    # image stage. one of ['bare', 'base', 'build', 'install']
+  provider: "" # cloud provider. only 'aws' is supported
+```
+
+## `sync_repos`
+
+`sync_repos` is an optional field which declares which repositories are to be
+synced and how they are to be synced. This field will add build stages to sync
+the repository as well as update the job description with details about what
+repos the job instance is being run with. The syntax is:
+
+```yaml
+sync_repos:
+  - name: "" # repository name under github.com/openshift
+    type: "" # 'pull_request' for a PR sync [optional]
+```
+
+## `actions`
+
+`actions` is an optional list of actions to take in the job. Each action will
+declare a type and can optionally declare a timeout. The syntax is:
+
+```yaml
+actions:
+  - type: ""   # one of ['forward_parameters', 'host_script', 'script']
+    timeout: 0 # in seconds [optional]
+```
+
+### `type: "forward_parameters"`
+
+the `forward_parameters` action will make an environment variable available on
+the Jenkins master in the `$WORKSPACE` also available to scripts running on the
+remote host in EC2. The syntax is:
+
+```yaml
+actions:
+  - type: "forward_parameters"
+    timeout: 0 # in seconds [optional]
+    parameters: [] # a list of environment variables
+```
+
+### `type: "host_script"`
+
+the `host_script` action will run a set of shell commands on the Jenkins master
+in the job's `$WORKSPACE`. The syntax is:
+
+```yaml
+actions:
+  - type: "host_script"
+    timeout: 0 # in seconds [optional]
+    title: ""  # a human-readable title for the step
+    script: "" # inline shell script to run, no character escaping necessary 
+```
+
+### `type: "script"`
+
+the `script` action will run a set of shell commands in the specified directory
+on the remote VM in EC2. The syntax is:
+
+```yaml
+actions:
+  - type: "script"
+    timeout: 0 # in seconds [optional]
+    title: ""      # a human-readable title for the step
+    repository: "" # the repository to `cd` into [optional]
```

Comment:
Need to let us know that this implies 'github.com/openshift' - can't be assumed just because sync_repos already mentions it.